portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,48 +29,48 @@
   private ArrayList<String> messages = new ArrayList<>();
 
   /**
-   * Write to /data the messages ArrayList as a json string
+   * Write to /data the messages ArrayList as a json string.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJson(messages);
+    String json = convertMessageToJson(messages);
     response.setContentType("text/html;");
     response.getWriter().println(json);
   }
 
   /**
-   * Obtain the input from the comment form and add it to the messages ArrayList
+   * Obtain the input from the comment form and add it to the messages ArrayList.
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the comment input from the form.
-    String comment = getParameter(request, "text-input", /* default value= */ "");
+    String comment = getParameter(request, "text-input").orElse("");
```

Comment:
Do you want to append an empty comment? Why not just skip adding the comment if it's empty? It'll be nice if you add some validation checks on the UI (e.g. disable the "Comment" button unless there's text in the text box), and probably don't append an empty comment here as well.

Reply:
Oops, I shouldn't be adding empty comments. I changed the doPost() function so that empty strings won't be added to the messages ArrayList. 

I also added a function that disables the submit button if the comment text area is empty.